src/evaluators/JSXElement.js

```diff
@@ -0,0 +1,298 @@
+      case "JSXAttribute":
+        let { name, value } = astAttribute;
+        invariant(name.type === "JSXIdentifier", `JSX attribute name type not supported: ${astAttribute.type}`);
+        attributes.set(name.name, evaluateJSXValue(value, strictCode, env, realm));
+        break;
+      case "JSXSpreadAttribute":
+        // TODO
+        return invariant(false, "TODO: JSX Spread Attibutes are not supported");
+      default:
+        invariant(false, `Unknown JSX attribute type:: ${astAttribute.type}`);
+    }
+  }
+  return {
+    attributes,
+    children,
+  };
+}
+
+function getReactElementSymbol(realm) {
+  if (reactElementSymbol !== undefined) {
+    return reactElementSymbol;
+  }
+  let SymbolFor = realm.intrinsics.Symbol.properties.get("for");
+  if (SymbolFor !== undefined) {
+    let SymbolForDescriptor = SymbolFor.descriptor;
+
+    if (SymbolForDescriptor !== undefined) {
+      let SymbolForValue = SymbolForDescriptor.value;
+      if (SymbolForValue !== undefined && typeof SymbolForValue.$Call === "function") {
+        reactElementSymbol = SymbolForValue.$Call(realm.intrinsics.Symbol, [
+          new StringValue(realm, reactElementSymbolKey),
+        ]);
+      }
+    }
+  }
+  return reactElementSymbol;
+}
+
+function createReactProps(realm, type, attributes, children, env) {
+  let obj = ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+  for (let [key, value] of attributes) {
+    if (typeof key === "string") {
+      if (RESERVED_PROPS.hasOwnProperty(key)) {
+        continue;
+      }
+      CreateDataPropertyOrThrow(realm, obj, key, value);
+    }
+  }
+  if (children !== null) {
+    CreateDataPropertyOrThrow(realm, obj, "children", children);
+  }
+  return obj;
+}
+
+function createReactElement(realm, type, key, ref, props) {
+  let obj = ObjectCreate(realm, realm.intrinsics.ObjectPrototype);
+  CreateDataPropertyOrThrow(realm, obj, "$$typeof", getReactElementSymbol(realm));
+  CreateDataPropertyOrThrow(realm, obj, "type", type);
+  CreateDataPropertyOrThrow(realm, obj, "key", key);
+  CreateDataPropertyOrThrow(realm, obj, "ref", ref);
+  CreateDataPropertyOrThrow(realm, obj, "props", props);
+  CreateDataPropertyOrThrow(realm, obj, "_owner", realm.intrinsics.null);
+  return obj;
+}
+
+export default function(
+  ast: BabelNodeJSXElement,
+  strictCode: boolean,
+  env: LexicalEnvironment,
+  realm: Realm
+): Value | Reference {
+  invariant(realm.react.enabled, "JSXElements can only be evaluated with the reactEnabled option");
+  let openingElement = ast.openingElement;
+  let type = evaluateJSXIdentifier(openingElement.name, strictCode, env, realm);
+  let { attributes, children } = evaluateJSXAttributes(
+    openingElement.name,
+    openingElement.attributes,
+    ast.children,
+    strictCode,
+    env,
+    realm
+  );
+  let key = attributes.get("key") || realm.intrinsics.null;
+  let ref = attributes.get("ref") || realm.intrinsics.null;
+
+  if (key === realm.intrinsics.undefined) {
```

Comment:
Above lines should take care of this.

Reply:
Users can still pass in undefined as a value for `key` or `ref`.